build-logic/third-party-installations/src/main/kotlin/gradlebuild.android-studio-test-setup.gradle.kts

```diff
@@ -0,0 +1,143 @@
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import extensions.AndroidStudioInstallationExtension
+
+repositories {
+    ivy {
+        // Url of Android Studio archive
+        url = uri("https://redirector.gvt1.com/edgedl/android/studio/ide-zips")
+        patternLayout {
+            artifact("[revision]/[artifact]-[revision]-[ext]")
+        }
+        metadataSources { artifact() }
+        content {
+            includeGroup("android-studio")
+        }
+    }
+}
+
+val os = System.getProperty("os.name").toLowerCase()
+val architecture = System.getProperty("os.arch").toLowerCase()
+fun isWindows(): Boolean = os.startsWith("windows")
+fun isMacOS(): Boolean = os.startsWith("mac")
+fun isLinux(): Boolean = os.startsWith("linux")
+fun isIntel(): Boolean = architecture == "x86_64" || architecture == "x86"
+
+val extension = extensions.create<AndroidStudioInstallationExtension>("androidStudioInstallation").apply {
+    autoDownloadAndroidStudio.convention(false)
+    runAndroidStudioInHeadlessMode.convention(false)
+}
+
+val androidStudioRuntime by configurations.creating
+
+dependencies {
+    val extension = when {
+        isWindows() -> "windows.zip"
+        isMacOS() && isIntel() -> "mac.zip"
+        isMacOS() && !isIntel() -> "mac_arm.zip"
+        isLinux() -> "linux.tar.gz"
+        else -> throw IllegalStateException("Unsupported OS: $os")
+    }
+    androidStudioRuntime("android-studio:android-studio@$extension")
+}
+
+androidStudioRuntime.withDependencies {
+    this.forEach { dependency ->
+        if (dependency is ExternalDependency && dependency.version == null) {
+            dependency.version { require(extension.androidStudioVersion.get()) }
+        }
+    }
+}
+
+val unpackAndroidStudio = tasks.register<Copy>("unpackAndroidStudio") {
+    from(
+        Callable {
+            val singleFile = androidStudioRuntime.singleFile
+            when {
+                singleFile.name.endsWith(".tar.gz") -> tarTree(singleFile)
+                else -> zipTree(singleFile)
+            }
+        }
+    )
+    into("$buildDir/android-studio")
+}
+
+val androidStudioInstallation = objects.newInstance<AndroidStudioInstallation>().apply {
+    studioInstallLocation.fileProvider(unpackAndroidStudio.map { it.destinationDir })
+}
+
+tasks.withType<Test>().configureEach {
+    jvmArgumentProviders.add(
+        AndroidStudioSystemProperties(
+            androidStudioInstallation,
+            extension.autoDownloadAndroidStudio,
+            extension.runAndroidStudioInHeadlessMode,
+            providers
+        )
+    )
+    // Propagate JAVA_HOME so Android Studio can find a JDK for Gradle
```

Comment:
Should we use the JDK we want to run the test with? Or the JDK we use for compiling? Or maybe a JDK with a fixed version?

Though I wouldn't propagate this `JAVA_HOME` as environment variable to the test, since the test already is configured to use a particular one. The test itself should make sure to set the right `JAVA_HOME` environment variable when invoking Android Studio.

Reply:
It makes sense that we use JDK we want to run the test with. 

Since we call profiler code directly it's a bit trickier to set JAVA_HOME. I think it would be best we fix gradle-profiler to allow setting Android Studio process env variables.